dbms/src/IO/WriteBuffer.h

```diff
@@ -35,7 +35,7 @@ class WriteBuffer : public BufferBase
       */
     inline void next()
     {
-        if (!offset())
+        if (!offset() && available())
```

Comment:
Don't understand why it haven't broken anything. `next()` is used to flush data in a lot of places.